python_modules/dagster-pipes/dagster_pipes/__init__.py

```diff
@@ -672,6 +722,169 @@ def load_context(self, params: PipesParams) -> Iterator[PipesContextData]:
             )
 
 
+class PipesOutErrLogWriter(
+    PipesLogWriter
+):  # log writers can potentially capture other type sof logs (for example, from Spark workers)
+    """Log writes which collect stdout and stderr of the current process should inherit from this class."""
+
+    LOGS_DIR_KEY = "logs_dir"
+
+    @contextmanager
+    def open(self, params: PipesParams) -> Iterator:
+        is_session_closed = threading.Event()
+        stdout_capturing_started = threading.Event()
+        stderr_capturing_started = threading.Event()
+
+        with ExitStack() as stack:
+            try:
+                stdout_channel = self.make_channel(params, stream="stdout")
+                stderr_channel = self.make_channel(params, stream="stderr")
+
+                stack.enter_context(
+                    stdout_channel.capture(stdout_capturing_started, is_session_closed)
+                )
+                stack.enter_context(
+                    stderr_channel.capture(stderr_capturing_started, is_session_closed)
+                )
+
+                # make sure capturing is working
+                stdout_capturing_started.wait()
+                stderr_capturing_started.wait()
+
+                time.sleep(0.1)  # make sure it's REALLY started
+
+                yield
+            finally:
+                is_session_closed.set()
+
+
+class PipesOutErrLogWriterChannel(PipesLogWriterChannel):
+    """A base class for log writer channels that capture stdout and stderr of the current process."""
+
+    def __init__(self, stream: Literal["stdout", "stderr"], interval: float, name: str):
```

Comment:
Why is a `name` required? Seems like we should be able to generate one from the `stream`.

Reply:
Because this class doesn't know where it's going to be writing the logs, it only knows how to read them. 
I'm thinking to maybe separate this functionality into 2 different components. 
It should be possible to configure logs collection from log writing independently.